examples/warp_subscriptions/src/main.rs

```diff
@@ -0,0 +1,213 @@
+//!
+//! This example demonstrates asynchronous subscriptions usage with warp.
+//! NOTE: this uses tokio 0.2.alpha
+//!
+
+use std::{pin::Pin, sync::Arc, time::Duration};
+
+use futures::{Future, FutureExt as _, Stream};
+use tokio::timer::Interval;
+use warp::{http::Response, Filter};
+
+use juniper::{EmptyMutation, FieldError, RootNode};
+use juniper_warp::playground_filter;
+
+#[derive(Clone)]
+struct Context {}
+
+impl juniper::Context for Context {}
+
+#[derive(juniper::GraphQLEnum, Clone, Copy)]
+enum UserKind {
+    Admin,
+    User,
+    Guest,
+}
+
+struct User {
+    id: i32,
+    kind: UserKind,
+    name: String,
+}
+
+//struct EmptyMutation {}
+
+//#[juniper::object(Context = Context)]
+//impl EmptyMutation {}
+
+#[juniper::object(Context = Context)]
+impl User {
+    fn id(&self) -> i32 {
+        self.id
+    }
+
+    fn kind(&self) -> UserKind {
+        self.kind
+    }
+
+    fn name(&self) -> &str {
+        &self.name
+    }
+
+    async fn friends(&self) -> Vec<User> {
+        if self.id == 1 {
+            return vec![
+                User {
+                    id: 11,
+                    kind: UserKind::User,
+                    name: "user11".into(),
+                },
+                User {
+                    id: 12,
+                    kind: UserKind::Admin,
+                    name: "user12".into(),
+                },
+                User {
+                    id: 13,
+                    kind: UserKind::Guest,
+                    name: "user13".into(),
+                },
+            ];
+        } else if self.id == 2 {
+            return vec![User {
+                id: 21,
+                kind: UserKind::User,
+                name: "user21".into(),
+            }];
+        } else if self.id == 3 {
+            return vec![
+                User {
+                    id: 31,
+                    kind: UserKind::User,
+                    name: "user31".into(),
+                },
+                User {
+                    id: 32,
+                    kind: UserKind::Guest,
+                    name: "user32".into(),
+                },
+            ];
+        } else {
+            return vec![];
+        }
+    }
+}
+
+#[derive(juniper::GraphQLObject)]
+#[graphql(description = "A humanoid creature in the Star Wars universe")]
```

Comment:
FYI, you can use normal rust doc comments here and they will be used for GraphQL